### ISSUE TITLE
[BUG] Socket closed

### ISSUE THREAD

**Issue by volodk** [zeapo/Android-Password-Store#1846]
### Describe the bug

Cannot clone/sync remote github repo.

### Steps to reproduce

1. Install latest version of the app
2. Generate new ssh key pair (ECDSA)
3. Import public key to github
4. Copy/paste git repo url to the app settings: git@github.com:username/.password-store.git
5. Try to clone remote repo and get an error message:
"An error occured during a Git operation  Socket closed"

### Expected behavior

Be able to clone remote git repo

### Device information

- Device: Samsung S20 FE
- OS: Android 12
- App version 1.13.5

### Additional context

I was a happy user of old version of the Password store app until github started to reject my ssh keys (rsa).

So I decided to upgrade the app.

**Comment by msfjarvis** [comment]
This is plainly a connection error, we cannot debug it without system logs. You can refer to https://github.com/android-password-store/Android-Password-Store/issues/1143 for a previous instance of such an issue to determine debugging steps.

**Comment by volodk** [comment]
Yep, guys, I regenerated ssh pair without that screen lock protection and everything started to work as expected.

**Comment by stephen304** [comment]
I also encountered this, tried both github and gitlab but it was regenerating the key that fixed it. I decided to still use screen lock and it seems to still work fine.

**Comment by tennox** [comment]
I can confirm this is still an issue (v1.13.5) and is fixed by generating ssh key without screen lock.

Found this in logcat:
```
02-20 05:14:47.716 E/keystore2(595): keystore2::error: In create_operation.
02-20 05:14:47.716 E/keystore2(595): Caused by:
02-20 05:14:47.716 E/keystore2(595):     0: In authorize_create: matching auth token is expired.
02-20 05:14:47.716 E/keystore2(595):     1: Error::Km(ErrorCode(-26))
```

**Comment by tennox** [comment]
Just tested on latest develop branch - same result :cry: 
( but sweet UI improvements :heart_eyes: )

**Comment by meliache** [comment]
I had the same issue with the 9eac07efc32bd3aca1075536e556c9036fc7ccb4 build on my Pixel Android 14, regenerating the key (with screen lock) worked.

What's weird, when I first stumbled upon this bug, my first reaction was to reset the app storage & cache, reinstall it and I then effectively regenerated an ssh key in the login screen. And I did this twice (trying free and non-free builds). But somehow it seems I needed to regenerate the key in the app after first having generated a different key, without resetting the cache inbetween :thinking: Maybe it's also just that I have to generate the key in the settings menu, not in the login wizard.
